exercises/queen-attack/src/lib.rs

```diff
@@ -1 +1,19 @@
+pub struct ChessPosition;
 
+pub struct Queen;
+
+impl ChessPosition {
+    pub fn new(rank: i32, file: i32) -> Result<Self, i32> {
+        unimplemented!("Construct a ChessPosition struct, given the following rank, file: ({}, {}). If the position is invalid return error that contains wrong rank/file.", rank, file);
+    }
+}
+
+impl Queen {
+    pub fn new(_position: ChessPosition) -> Self {
+        unimplemented!("Given the chess position, construct a Queen struct.");
```

Comment:
Please include the `position` variable so the leading underscore is unnecessary: `"Given the chess position {:?}, construct a Queen struct."`. This will require a `#[derive(Debug)]` annotation on `ChessPosition`.